buildkite/resource_pipeline_team_test.go

```diff
@@ -0,0 +1,226 @@
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV6ProviderFactories: protoV6ProviderFactories(),
+		CheckDestroy:             testCheckPipelineTeamResourceRemoved,
+		Steps: []resource.TestStep{
+			{
+				Config:             testAccPipelineTeamConfigBasic(teamName, "READ_ONLY"),
+				ExpectNonEmptyPlan: true,
+				Check: resource.ComposeAggregateTestCheckFunc(
+					// Confirm the test resource team & team pipeline exists in the Buildkite API
+					testAccCheckPipelineTeamExists("buildkite_pipeline_team.pipelineteam", &tp),
+					// Confirm the team has the correct values in Buildkite's system
+					testAccCheckPipelineTeamRemoteValues("READ_ONLY", &tp),
+					// Confirm the team pipeline has the correct values in terraform state
+					resource.TestCheckResourceAttr("buildkite_pipeline_team.pipelineteam", "access_level", "READ_ONLY"),
+				),
+			},
+			{
+				Config:             testAccPipelineTeamConfigBasic(teamNameNew, "MANAGE_BUILD_AND_READ"),
+				ExpectNonEmptyPlan: true,
+				Check: resource.ComposeAggregateTestCheckFunc(
+					// Confirm the test resource team & team pipeline exists in the Buildkite API
+					testAccCheckPipelineTeamExists("buildkite_pipeline_team.pipelineteam", &tp),
+					// Confirm the team has the correct values in Buildkite's system
+					testAccCheckPipelineTeamRemoteValues("MANAGE_BUILD_AND_READ", &tp),
+					// Confirm the team pipeline has the correct values in terraform state
+					resource.TestCheckResourceAttr("buildkite_pipeline_team.pipelineteam", "access_level", "MANAGE_BUILD_AND_READ"),
+				),
+			},
+		},
+	})
+}
+
+// Confirm that this resource can be imported
+func TestAccPipelineTeam_Import(t *testing.T) {
+	var tp pipelineTeamResourceModel
+	teamName := acctest.RandString(12)
+	t.Parallel()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV6ProviderFactories: protoV6ProviderFactories(),
+		CheckDestroy:             testCheckPipelineTeamResourceRemoved,
+		Steps: []resource.TestStep{
+			{
+				Config:             testAccPipelineTeamConfigBasic(teamName, "READ_ONLY"),
+				ExpectNonEmptyPlan: true,
+				Check: resource.ComposeAggregateTestCheckFunc(
+					// Confirm the test resource team & team pipeline exists in the Buildkite API
+					testAccCheckPipelineTeamExists("buildkite_pipeline_team.pipelineteam", &tp),
+					// Confirm the team has the correct values in Buildkite's system
+					resource.TestCheckResourceAttrSet("buildkite_pipeline_team.pipelineteam", "team_id"),
+					resource.TestCheckResourceAttrSet("buildkite_pipeline_team.pipelineteam", "pipeline_id"),
+					resource.TestCheckResourceAttr("buildkite_pipeline_team.pipelineteam", "access_level", "READ_ONLY"),
+				),
+			},
+			{
+				// re-import the resource and confirm they match
+				ResourceName:      "buildkite_pipeline_team.pipelineteam",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
```

Comment:
I think it would be good to have a test where terraform creates the pipeline_team but then its removed on the API side. To ensure that it removes it from state and plans to recreate it